### PR TITLE
breaking: publish esm (fixes #352)

### PR DESCRIPTION
**Pull request by aladdin-add**
the main changes are:
* drop node < 12
* convert index.js => esm
* pkg.type => "module"
* upgrade to ava v3
* update examples in docs

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -50,5 +51,11 @@
 		"fast-check": "^1.5.0",
 		"tsd": "^0.7.3",
 		"xo": "^0.24.0"
+	},
+	"xo": {
+		"rules": {
+			"unicorn/import-index": 0,
+			"import/extensions": 0
+		}
```
Comment (aladdin-add) [inline package.json:59]: this is required as ava does not fully support esm.

```bash
"Error [ERR_UNSUPPORTED_DIR_IMPORT]"
```

you can remove it after ava esm supports.

**Comment by sindresorhus** [comment]
I appreciate the PR, but I don't intend to convert this package u til later this year (it will cause too much support). I also prefer to do it myself as it will take much more work to comment on what needs changing than just doing the conversion myself.

**Comment by aladdin-add** [comment]
do you mind add `"type": "commonjs"` or `"main": "index.js"`, to help tools better understanding it's a cjs module?

**Comment by sindresorhus** [comment]
No, tools are broken if they don't already assume it's CJS when there's to "type:module".

**Comment by aladdin-add** [comment]
well, it also makes sense. 👍

**Pull request closed without merging**